pkgs/development/tools/just/default.nix

```diff
@@ -38,22 +38,25 @@ rustPlatform.buildRustPackage rec {
     sed -i src/justfile.rs \
         -i tests/*.rs \
         -e "s@/bin/echo@${coreutils}/bin/echo@g" \
-        -e "s@#!/usr/bin/env sh@#!${bash}/bin/sh@g" \
         -e "s@#!/usr/bin/env cat@#!${coreutils}/bin/cat@g" \
-        -e "s@#!/usr/bin/env bash@#!${bash}/bin/sh@g"
+        -e "s@#!/usr/bin/env sh@#!${bash}/bin/sh@g" \
+        -e "s@#!/usr/bin/env bash@#!${bash}/bin/bash@g"
 
     # Return unchanged string.rs
     cp $TMPDIR/string.rs tests/string.rs
   '';
 
-  # Skip "edit" when running "cargo test", since this test case needs "cat" and "vim".
-  # Skip "choose" when running "cargo test", since this test case needs "fzf".
-  checkFlags = [ "--skip=choose" "--skip=edit" ];
+  checkFlags = [
+    "--skip=choose" # test case needs "cat" and "vim"
+    "--skip=edit" # test case needs "fzf"
```

Comment:
Can we add fzf, coreutils and vim to checkInputs?

Reply:
yeah I can try add them in. Is there a concern that it'd bloat the closure size or is it fine as they're added through checkInputs?

Reply:
checkInputs shouldn't bloat the closure size. Do we need to patch paths in just to make fzf work?

Reply:
So `Command` isn't able to run `vim` even though I've verified it's in the PATH. I'm guessing it falls back to use `/usr/bin/env` or `which` to find where `vim` is (which doesn't exist in the sandbox).
This isn't an issue when using `just` outside of the sandbox. It cant be resolved by patching the test and I don't really want to be patching the source code just for the check phase with: `"s@.unwrap_or_else(|| \"vim\".into());@.unwrap_or_else(|| \"${coreutils}/bin/cat\".into());@g" (patching it for `${vim}/bin/vim` just hangs)`, it's probably best to just skip `edit` with more detailed explanation

https://github.com/casey/just/blob/0a9ffd7a7b1e423e8ae5cee08427a416c957644f/src/config.rs#L686-L691

It's a similar problem for the chooser not being able to find `fzf`. It can be resolved with `export JUST_CHOOSER="${coreutils}/bin/cat"` as there's no test that forces the run of the fallback value.

(As a note if I feed it `export JUST_CHOOSER="${fzf}/bin/fzf"` it doesn't like it and fails with error code 2 `stderr: inappropriate ioctl for device`, probably also due to the sandbox too?)